### DISCUSSION TITLE
How can we simulate the masked out part of the arc to get not only lower reduced chi squared but also a better fit as model?

### DISCUSSION THREAD

**Discussion by rafee-adnan** [AstroBridge/BDLensing#38] [Q&A]
I finally got a reduced chi squared value of 1.53 for my system. But I had to mask out some parts of the lens arc to reach that value, which I think is not the right way. As I think, the method we are using right now do not simulate the mass model of the lensed galaxy. How can I simulate the lensed galaxy properly without masking out vital parts of the arc?
So what should be my next step? (I am attaching my simulated result and the observed data for better comparison.)
[DESIJ0201_observed_data.pdf](https://github.com/ajshajib/BDLensing/files/11274102/DESIJ0201_observed_data.pdf)
<img width="886" alt="DESIJ0201_modeled_data" src="https://user-images.githubusercontent.com/76589056/233105605-c0c6e63a-80f2-487d-a3ea-834a9d611455.png">

**Comment by rafee-adnan** [comment]
I figured out a way to increase the accuracy of the model by increasing `n_max` or the number of shapelets. I got the lowest reduced chi-squared value 1.02 for `n_max=12`  (2nd image below: n_sersic=2, n_max=12). And I ran several simulations varying `n_sersic`, `n_max` and `supersampling_factor`. The most notable change was increasing and decreasing the amount of `shapelets(n_max).` Now sometimes the same number and settings returns two set of different images of reconstructed source, convergence, and magnification model (1st image: n_sersic=2, n_max=10, 2nd & 3rd image: n_sersic=2, n_max=12). They seem zoomed in or out. Why is it happening, and how can I return the same magnification every time? The images in your paper, Shajib et al. (2021) all have similar magnification. 

Another thing I would like to mention is the problem of overfitting in the wrong direction. The shape of the reconstructed source varied a lot with the lens image [Shajib et al. (2021)]. But In my case if you look carefully the reconstructed source resembles the arc in the lens image. Is it an issue? How can I resolve this issue?

<img width="893" alt="n_sersic, n_max, 2, 10" src="https://user-images.githubusercontent.com/76589056/233807084-121064df-f7e3-4298-8a45-af4cf06a127e.png">
<img width="883" alt="1  n_sersic, n_max, 2, 12" src="https://user-images.githubusercontent.com/76589056/233807094-6330e7f9-a7de-4445-86fa-8b73247c2151.png">
<img width="888" alt="2  n_sersic, n_max, 2, 12" src="https://user-images.githubusercontent.com/76589056/233807097-48b18fa2-7386-42c1-8030-1f874086991b.png">

**Reply by ajshajib** [reply]
Yes, the reconstructed source should not look like the arc, then the model is basically trying to get a good residual without having to do any lensing (`theta_E` will become very small in that case). What's the upper limit for source `R_sersic` and `beta`? You can try setting the upper limit to something like 0.2". You can also have `n_sersic` not fixed and sampled. If `theta_E` still goes to a very small value, you can also set a lower limit for it, looking at the image, 0.8"  looks to be a reasonable lower limit.

**Reply by rafee-adnan** [reply]
The upper limit and lower limit for source `R_sersic` are 10 and 0.001. And the upper limit and lower limit for source `beta` are 100 and 0.01. Should I set the upper limit to 0.2 for both of them? Or should I set the lower limit to 0.2? Because the value is converging to a lower value, source `R_sersic= 0.002` and `beta= 0.1-0.05` & `theta_E= 0.7-0.5`.

**Reply by ajshajib** [reply]
Set the upper limit to 0.2" for both.
